common/djangoapps/rocket_chat/api/v0/views.py

```diff
@@ -0,0 +1,76 @@
+import logging
+
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework import permissions
+from rest_framework import status
+
+from opaque_keys.edx.keys import CourseKey
+
+from rocket_chat.utils import (
+    get_rocket_chat_settings,
+    initialize_api_rocket_chat,
+    create_user,
+    get_subscriptions_rids,
+)
+from .serializers import RocketChatCredentialsSerializer
+
+LOG = logging.getLogger(__name__)
+
+
+class RocketChatCredentials(APIView):
+
+    authentication_classes = (
+        SessionAuthentication,
+    )
+    permission_classes = (permissions.IsAuthenticated,)
+
+    def get(self, request, course_id):
+        """
+        This returns a Json with the  user credentials in order to use the rocketchat api methods outside of the server
+        """
+        user = request.user
+
+        course_key = CourseKey.from_string(course_id)
+
+        rocket_chat_settings = get_rocket_chat_settings()
+
+        api_rocket_chat = initialize_api_rocket_chat(rocket_chat_settings)
+
+        if api_rocket_chat:
+
+            user_info = api_rocket_chat.users_info(username=user.username)
+
+            try:
+                user_info = user_info.json()
+                if not user_info.get('success', False):
+                    create_user(api_rocket_chat, user, course_key)
+            except AttributeError:
+                LOG.error("Rocketchat API response with: %s", user_info)
```

Comment:
PLease stop logging user details in my repos!!!! this is not cool! loging user information is a break in general security practices!